### ISSUE TITLE
Synchronous request and response ?

### ISSUE THREAD

**Issue by myfreescalewebpage** [RIAEvangelist/node-ipc#158]
Hello all,

First wanted to say I'm new to Node.JS and try to learn by practice, so maybe my question is trivial, but anyway I have google a lot and haven't found a clear solution to this issue.

I have basically followed https://github.com/RIAEvangelist/node-ipc#basic-examples to get a Unix Sockets example with server+client exchanging messages. Working.

In the application I'm developing, I try to send a message to the IPC server and get an answer to serve an HTTP request. An extract of the code is below, I'm using 'express'.

```
var express = require('express');
var router = express.Router();
router.get('/data', ensureAuthenticated, function(req, res) {
    /* I want to send a request to the IPC server here and return the response back to the HTTP client */
    res.json(response_from_ipc_server);
});
```

Basically, this is simply sending a message and receiving another one, but _synchronously_.

How to achieve this with node-ipc ?

Thanks,
Joel

**Comment by RIAEvangelist** [comment]
in the docs there is a section on synch search for syn in the docs you'll find it. Same in the examples as well.

**Comment by joshuaalayon** [comment]
Be aware - this package was found to contain malware that detects its users machines purported geographical location and replaces the data from the machine if it does not like the location it detects

If you are still using this package - you may do better by locking it to an older version or, preferably, replacing it altogether along with all software maintained by the repos maintainer(s)

https://www.bleepingcomputer.com/news/security/big-sabotage-famous-npm-package-deletes-files-to-protest-ukraine-war/amp/
